### PR TITLE
added a new endpoint

### PR DESCRIPTION
**Pull request by sacintha**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/data/jsonprofile/CIAGProfileOffenderIdListRequestDTO.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.hmpps.hmppsciagcareersinductionapi.data.jsonprofile
+
+import io.swagger.v3.oas.annotations.media.Schema
+import javax.validation.constraints.Size
+
+data class CIAGProfileOffenderIdListRequestDTO(
+  @Schema(description = "This is the ID of the inmate ", name = "offenderIds", pattern = "^[A-Z]\\d{4}[A-Z]{2}\$", required = true)
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/data/jsonprofile/CIAGProfileOffenderIdListRequestDTO.kt:7]: IDs (plural). Perhaps change to something like `A list of at least one or more Offender IDs`.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt [hunk 2]
```diff
@@ -81,6 +83,51 @@ class CIAGResourceController(
     return ciagProfileService.getCIAGProfileForOffender(offenderId)
   }
 
+  @PreAuthorize("hasAnyRole('ROLE_EDUCATION_WORK_PLAN_EDITOR','ROLE_EDUCATION_WORK_PLAN_VIEWER')")
+  @PostMapping("/list")
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt:87]: So you're sticking with `/ciag/induction/list` then? As per my example with `/books` on your other PR, having a base path of say `/ciag/inductions` would have been more RESTful in my opinion.

File: src/test/resources/db.h2/V1_1__start.sql [hunk 3]
```diff
@@ -0,0 +1,68 @@
+drop table IF EXISTS ability_to_work_impact;
```
Comment (mattwills-moj-digital) [inline src/test/resources/db.h2/V1_1__start.sql:1]: I don't understand why you need these scripts in `src/test/resources`? What is different about them from the main postgres ones that stops them working in h2?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt [hunk 4]
```diff
@@ -81,6 +83,51 @@ class CIAGResourceController(
...
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = String::class))],
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Invalid Parameters have been passed",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Resource not found",
```
[the remark this answers is not in the document]
Comment (sacintha) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt]: There should be not a scenario where there is no resource being sent , atleast an empty list would be send . so  removing the 404

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/data/jsonprofile/CIAGProfileOffenderIdListRequestDTO.kt (hunk 1 above)
Comment (sacintha) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/data/jsonprofile/CIAGProfileOffenderIdListRequestDTO.kt]: changed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt (hunk 2 above)
Comment (sacintha) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt:87]: In my opinion both are valid the base url tells what resource we are sending and the list just tells me that it is a list .
I do not think that should be a major issue

Comment (theDustRoom) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsciagcareersinductionapi/resource/CIAGResourceController.kt:87]: I'm happy with this for now, we can revisit this later if necessary

**Review by theDustRoom — DISMISSED** [review]
I'm happy with this now

**Review by theDustRoom — APPROVED** [review]

**Review by theDustRoom — APPROVED** [review]

**Review by theDustRoom — DISMISSED** [review]

**Review by theDustRoom — APPROVED** [review]
Try again...

**Pull request merged**